com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/PropertiesUtil.java

```diff
@@ -138,4 +141,15 @@ public static String transformResourcePropertyValue(String value) {
         return resources;
     }
 
+    public static void createResourcePropertyTextureTask(Project project, IResource resource, IResource input) throws CompileExceptionError {
+        if (project.getBuilderFromExtension(resource) == TextureBuilder.class) {
+            Task<?> embedTask = project.createTask(resource, TextureBuilder.class);
```

Comment:
As I mentioned, what about other property types?

Also, I feel that the code should be possible to rewritten as:

```
klass = project.getBuilderFromExtension(resource);
task = project.createTask(resource, klass)
```

And going even further, why not then have a `project.createTask(resource)` ?

Reply:
>And going even further, why not then have a project.createTask(resource) ?

We have it, but in this case I don't use it because some resources may have different builders. In this case I know what exactly builder I need and I want to specify it.

>As I mentioned, what about other property types?

At the moment it doesn't make sense because we build all the resources anyways (except textures). It will be a separate task to check every resource one by one.